kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -423,13 +424,16 @@ private static void validatePhysicalNameConsistency(
 
   /* Validate if a given schema evolution is safe for a given column mapping mode*/
   private static void validateSchemaEvolution(
-      StructType currentSchema, StructType newSchema, Metadata metadata) {
+      StructType currentSchema,
+      StructType newSchema,
+      Metadata metadata,
+      Set<String> clusteringColumnPhysicalNames) {
     ColumnMappingMode columnMappingMode =
         ColumnMapping.getColumnMappingMode(metadata.getConfiguration());
     switch (columnMappingMode) {
       case ID:
       case NAME:
-        validateSchemaEvolutionById(currentSchema, newSchema);
+        validateSchemaEvolutionById(currentSchema, newSchema, clusteringColumnPhysicalNames);
```

Comment:
Add a comment at the place where you block schema change if column mapping is disabled. Just incase if we remove that constraint in future to handle the clustering columns in schema evolution (e.g. adding a new clustering column).

Reply:
I added a comment, though I think the main thing to consider with regards to clustering columns in case the column mapping restraint is removed is that we won't have the ability to properly validate changes since we won't have physical names. I added a separate ToDo comment for supporting adding a new clustering columns.